pkg/logql/log/parser.go

```diff
@@ -486,3 +500,76 @@ func (u *UnpackParser) unpack(it *jsoniter.Iterator, entry []byte, lbs *LabelsBu
 	}
 	return entry, nil
 }
+
+type SyslogParser struct {
+	maxMessageLength int
+	lbs              *LabelsBuilder
+}
+
+func NewSyslogParser() *SyslogParser {
+	return &SyslogParser{maxMessageLength: 1 << 12}
+}
+
+func (l *SyslogParser) Process(line []byte, lbs *LabelsBuilder) ([]byte, bool) {
+	if lbs.ParserLabelHints().NoLabels() {
+		return line, true
+	}
+	l.lbs = lbs
+	b := line[0]
+	r := bytes.NewReader(line)
+	if b == '<' {
+		nontransparent.NewParser(
+			syslog.WithListener(l.handleMessage),
```

Comment:
syslog.WithListener(l.handleMessage) could also be part of the struct for reuse.